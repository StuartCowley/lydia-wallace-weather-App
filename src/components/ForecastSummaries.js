import "../styles/ForecastSummaries.css";

import React from "react";

import PropTypes from "prop-types";
import ForecastSummary from "./ForecastSummary";

function ForecastSummaries(props) {
  const { forecasts, setSelectedDate } = props;
  return (
    <div className="forecast-summaries">
      {forecasts.map((forecast) => (
        <ForecastSummary
          key={forecast.date}
          date={forecast.date}
          temperature={forecast.temperature}
          description={forecast.description}
          icon={forecast.icon}
          setSelectedDate={setSelectedDate}
        />
      ))}
    </div>
  );
}

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      temperature: PropTypes.shape({
        max: PropTypes.number,
        min: PropTypes.number,
      }),
      wind: PropTypes.shape({
        speed: PropTypes.number,
        direction: PropTypes.string,
      }),
      humidity: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.number,
    })
  ).isRequired,
  setSelectedDate: PropTypes.func.isRequired,
};

export default ForecastSummaries;
