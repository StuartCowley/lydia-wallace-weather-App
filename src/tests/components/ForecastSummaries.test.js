import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = [
    {
      date: 1525046400000,
      temperature: {
        max: 15,
        min: 10,
      },
      wind: {
        speed: 11,
        direction: "se",
      },
      humidity: 67,
      description: "test",
      icon: 800,
    },
    {
      date: 1525046400022,
      temperature: {
        max: 10,
        min: 5,
      },
      wind: {
        speed: 19,
        direction: "n",
      },
      humidity: 78,
      description: "test2",
      icon: 800,
    },
  ];
  const setterProp = { setSelectedDate: () => {} };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProps}
        setSelectedDate={setterProp.setSelectedDate}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries
        forecasts={validProps}
        setSelectedDate={setterProp.setSelectedDate}
      />
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
