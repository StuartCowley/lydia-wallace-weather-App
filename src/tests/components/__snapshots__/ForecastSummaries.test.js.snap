// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <b>
        <div
          class="forecastSummary__date"
        >
          Mon Apr 30 2018
        </div>
      </b>
      <div
        class="forecastSummary__temperature"
      >
        15°C
      </div>
      <div
        class="forecastSummary__description"
      >
        test
      </div>
      <div
        class="forecastSummary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny wi-flip-horizontal wi-rotate-90"
          name="owm"
        />
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <b>
        <div
          class="forecastSummary__date"
        >
          Mon Apr 30 2018
        </div>
      </b>
      <div
        class="forecastSummary__temperature"
      >
        10°C
      </div>
      <div
        class="forecastSummary__description"
      >
        test2
      </div>
      <div
        class="forecastSummary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny wi-flip-horizontal wi-rotate-90"
          name="owm"
        />
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
